splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java

```diff
@@ -35,35 +34,40 @@
     static final CurrentNetwork NO_NETWORK = new CurrentNetwork(NetworkState.NO_NETWORK_AVAILABLE, null);
     static final CurrentNetwork UNKNOWN_NETWORK = new CurrentNetwork(NetworkState.TRANSPORT_UNKNOWN, null);
 
-    private final ConnectionMonitor connectionMonitor;
+    private final Supplier<NetworkRequest> createNetworkMonitoringRequest;
     private final NetworkDetector networkDetector;
+    private final ConnectivityManager connectivityManager;
 
-    private final AtomicReference<CurrentNetwork> currentNetwork = new AtomicReference<>();
+    private volatile CurrentNetwork currentNetwork;
+    private volatile ConnectionStateListener connectionStateListener;
 
     ConnectionUtil(Context context) {
         this(ConnectionUtil::createNetworkMonitoringRequest, NetworkDetector.create(context), (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
     }
 
     //for testing, since building the NetworkRequest fails in junit due to .. Android.
     ConnectionUtil(Supplier<NetworkRequest> createNetworkMonitoringRequest, NetworkDetector networkDetector, ConnectivityManager connectivityManager) {
-        this.connectionMonitor = new ConnectionMonitor();
-
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            connectivityManager.registerDefaultNetworkCallback(connectionMonitor);
-        } else {
-            NetworkRequest networkRequest = createNetworkMonitoringRequest.get();
-            connectivityManager.registerNetworkCallback(networkRequest, connectionMonitor);
-        }
+        this.createNetworkMonitoringRequest = createNetworkMonitoringRequest;
         this.networkDetector = networkDetector;
+        this.connectivityManager = connectivityManager;
         refreshNetworkStatus();
```

Comment:
I think this call should be moved down to `start`. See my implementation in #146  :)

Reply:
also note that you don't need to keep references to them if they just get passed into the `start` method with my approach.